tests/e2e/e2e_suite_test.go

```diff
@@ -168,37 +184,43 @@ func (s *IntegrationTestSuite) upgrade() {
 	err = s.upgradeManager.ExportState(buildDir)
 	s.Require().NoError(err, "can't export node container state to local")
 
-	s.T().Log("killing initial node...")
+	s.T().Log("killing node before upgrade...")
 	err = s.upgradeManager.KillCurrentNode()
 	s.Require().NoError(err, "can't kill current node")
 
 	s.T().Logf(
-		"starting upgraded node: version: [%s] mount point: [%s]",
-		s.upgradeParams.TargetVersion,
+		"starting upgraded node: repo: [%s] version: [%s] mount point: [%s]",
+		targetRepo,
+		targetVersion,
 		s.upgradeParams.MountPath,
 	)
 
-	node := upgrade.NewNode(s.upgradeParams.TargetRepo, s.upgradeParams.TargetVersion)
+	node := upgrade.NewNode(targetRepo, targetVersion)
 	node.Mount(s.upgradeParams.MountPath)
 	node.SetCmd([]string{"evmosd", "start", fmt.Sprintf("--chain-id=%s", s.upgradeParams.ChainID)})
 	err = s.upgradeManager.RunNode(node)
 	s.Require().NoError(err, "can't mount and run upgraded node container")
 
-	s.T().Log("node started! waiting for node to produce 10 blocks")
+	s.T().Logf("node started! waiting for node to produce %d blocks", blocksAfterUpgrade)
 	// make sure node produce blocks after upgrade
-	err = s.upgradeManager.WaitForHeight(ctx, firstUpgradeHeight+10)
-	s.Require().NoError(err, "node not produce blocks")
+	err = s.upgradeManager.WaitForHeight(ctx, int(s.upgradeManager.UpgradeHeight)+blocksAfterUpgrade)
+	s.Require().NoError(err, "node does not produce blocks after upgrade")
```

Comment:
Add a sanity check to query the node's version?

Reply:
I don't think it's necessary because the node wouldn't start after the upgrade if the version was not the expected one. One would also have to add separate checks for actual version tags and then the "latest" local build